### PR TITLE
feat: [Logical Optimizer]Eliminate Join Rule

### PR DESCRIPTION
**Pull request by AveryQi115**
# Major Changes
Add an eliminate join rule which 
1. converts inner join to cross join when filter always equals to true
2. converts inner join to empty relation (remove the join) when filter always equals to false

## Rule Type
Heuristics(always apply), Transformation Rule(logical to logical)

# TODO
- [x] Create Empty Relation Support
- [x] Integrate with cross join type without condition

### CODE REVIEW

**Comment by AveryQi115** [comment]
@yliang412 yuchen, could you check whether the pr has any conflicting logic with your cross join support and arguments for logical optimizer? If no conflict, I'll merge it tonight.

**Comment by yliang412** [comment]
Doing some check right now

File: optd-datafusion-bridge/src/from_optd.rs [hunk 1]
```diff
@@ -345,7 +345,7 @@ impl OptdPlanContext<'_> {
         }
 
         let join_type = match node.join_type() {
-            JoinType::Inner => datafusion::logical_expr::JoinType::Inner,
+            JoinType::Inner | JoinType::Cross => datafusion::logical_expr::JoinType::Inner,
```
Comment (yliang412) [inline optd-datafusion-bridge/src/from_optd.rs]: `CrossJoin` already handled on line 343?

**Review by yliang412 — APPROVED** [review]
Overall LGTM

File: optd-datafusion-bridge/src/from_optd.rs (hunk 1 above)
Comment (AveryQi115) [reply optd-datafusion-bridge/src/from_optd.rs]: Removed the Cross in this branch. Thank you for catching it!

**Pull request merged**